terraform/interpolate.go

```diff
@@ -129,15 +129,6 @@ func (i *Interpolater) valueModuleVar(
 	n string,
 	v *config.ModuleVariable,
 	result map[string]ast.Variable) error {
-	// If we're computing all dynamic fields, then module vars count
-	// and we mark it as computed.
-	if i.Operation == walkValidate {
-		result[n] = ast.Variable{
-			Value: config.UnknownVariableValue,
-			Type:  ast.TypeString,
-		}
-		return nil
-	}
```

Comment:
What I'd do here is walk back to the PR / commit where this stanza was introduced and determine what the intent was just to double check that we're not reversing some prior decision.

The test suite should cover us, so the fact that everything is passing is probably enough, but I think for such a subtle area of the code base it's worth it to do extra due diligence.

Let me know what the git archeology turns up!